### PR TITLE
Add Spatial Sound extension

### PR DESCRIPTION
**Pull request by arthuro555**
This PR adds a spatial sounds extension that uses the howler 3d sound extension.
It also 
- Updates howler.min.js to v2.2.1
- Use howler core only in GDJS
- Add an example showing how to use it (fixtures not regenerated)

Note that spatial sound is only available with the WebAudio API, so music (html5 audio) cannot use this extension, only sounds.

### CODE REVIEW

**Comment by arthuro555** [comment]
Closes #1955

File: newIDE/app/resources/examples/spatial-sound/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+A small similar to the Lost Woods of The Legend of Zelda Ocarina of Time, that uses spatial sound.
```
Comment (4ian) [inline newIDE/app/resources/examples/spatial-sound/README.md:1]: missing a word here I think ;)

**Comment by 4ian** [comment]
Thanks for the PR :)

> Updates howler.min.js to v2.2.1
> Use howler core only in GDJS

Do you know if there are interesting bug fixes in this version?

> Note that spatial sound is only available with the WebAudio API, so music (html5 audio) cannot use this extension, only sounds.

In what case this applies? I can see the bug reports coming in already because of this 😬 Can we make this obvious to the user?

**Comment by arthuro555** [comment]
> Do you know if there are interesting bug fixes in this version?

If I remember correctly it fixes wav support on Safari 14 and an edge case causing an infinite loop

> Can we make this obvious to the user?

The way I have done it is simply exposing only an action for sounds and not an action for music, but I guess we can add a warning on the action?

**Comment by 4ian** [comment]
> The way I have done it is simply exposing only an action for sounds and not an action for music

I missed this, this sounds good then!

File: Extensions/SpatialSound/JsExtension.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      ),
+      'Arthur Pacaud (arthuro555)',
+      'MIT'
+    );
+
+    extension
+      .addAction(
+        'SetSoundPosition',
+        _('Set position of sound'),
+        _('Sets the spatial position of a sound.'),
```
Comment (4ian) [inline Extensions/SpatialSound/JsExtension.js:42]: Can we write in the description something about the units of the position? Like if I'm at pos 0,0,0 and the sounds is at 2,0,0, will I hear the sound? Basically, "how far" is "1"? :)

File: Extensions/SpatialSound/JsExtension.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  createExtension: function (
+    _ /*: (string) => string */,
+    gd /*: libGDevelop */
+  ) {
+    const extension = new gd.PlatformExtension();
+    extension.setExtensionInformation(
+      'SpatialSound',
+      _('Spatial sound'),
+      _(
+        'Allow positioning sounds in 3D space, using a stereo system to make the player know where a sound came from.'
```
Comment (4ian) [inline Extensions/SpatialSound/JsExtension.js:32]: I would say:
"Allow positioning sounds in a 3D space. The stereo system of the device is used to simulate the position of the sound and to give the impression that the sound is located somewhere around the player."

**Review by 4ian — APPROVED** [review]
A few minor changes/precisions and we should be good to go :)

File: Extensions/SpatialSound/JsExtension.js (hunk 2 above)
Comment (arthuro555) [reply Extensions/SpatialSound/JsExtension.js:42]: I am not sure to be honest. All I could find about WebAudio spatialization is [this resource](https://www.w3.org/TR/webaudio/#Spatialization-panning-algorithm), and it doesn't make it very clear at which point you wouldn't hear anymore (unless I overread something)

Comment (4ian) [reply Extensions/SpatialSound/JsExtension.js:42]: See this interesting article: https://medium.com/@kfarr/understanding-web-audio-api-positional-audio-distance-models-for-webxr-e77998afcdff

The attenuation follows an *inverse distance model*, meaning that by default the sound will be reduce by two at a distance of 2, and divided by 2 again at a distance of 4.

**Comment by 4ian** [comment]
Merged, thanks 👍

**Pull request merged**